pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -881,8 +908,9 @@ func (os *OpenStack) getVolume(diskName string) (volumes.Volume, error) {
 	return volume, err
 }
 
+// getComputeIDbyHostname tries to find ID of the OpenStack instance where
+// this code runs.
 func (os *OpenStack) getComputeIDbyHostname(cClient *gophercloud.ServiceClient) (string, error) {
```

Comment:
I had originally written it that way because server name can be different from hostname.
For ex in our env: Name is `kubernetes-master-3`, hostname is `kubernetes-master-3-6050`. 

Later I added the name check to it cause I think @jsafrane you said that hostname in your env was empty however your server names matched the hostnames.